### PR TITLE
NEW : External contact filter on contract list

### PR DESCRIPTION
**Pull request by ATM-Nicolas**
# New external contact filter on contract list

This will add a new filter on contract list to filter on contract's external contact :

image

### CODE REVIEW

File: htdocs/contrat/list.php [hunk 1]
```diff
@@ -433,6 +437,15 @@
 	$moreforfilter.=$form->selectarray('search_product_category', $cate_arbo, $search_product_category, 1, 0, 0, '', 0, 0, 0, 0, 'maxwidth300', 1);
 	$moreforfilter.='</div>';
 }
+// If the user can view contract type contact
+if ($conf->contrat->enabled && $user->rights->contrat->lire && $user->rights->service->lire)
+{
+	$moreforfilter.='<div class="divsearchfield">';
+	$moreforfilter.=$langs->trans('LinkedToSpecificThirdParty'). ': ';
```
Comment (eldy) [inline htdocs/contrat/list.php:443]: Is is to search contract linked to
- a third party
- contact of thirdparty
- or a type of contact of thirdparty ?

Comment (ATM-Nicolas) [reply htdocs/contrat/list.php:443]: To search contract linked to a type of contact of thirdparty

File: htdocs/langs/en_US/main.lang [hunk 2]
```diff
@@ -773,6 +773,7 @@ ByMonth=By month
 ByDay=By day
 BySalesRepresentative=By sales representative
 LinkedToSpecificUsers=Linked to a particular user contact
+LinkedToSpecificThirdParty=Linked to a particular thirdparty contact
```
Comment (eldy) [inline htdocs/langs/en_US/main.lang]: There is several pb with that:

1) You should use instead a label of combo that is
LinkedToAThirdPartyContactThatHasTheType=Linked to at least one thirdparty contact that has type ...

2) If contract is linked to 2 contacts C1 and C2 with type XXX, the number of lines in list will be duplicated.

3) It is a very strange need, so you should introduce a hidden option for that such a strange feature should not be provided by default to end users.

**Comment by eldy** [comment]
A way to make filter will be introduced into future version of Dolibarr and this PR will not be compatible with it so i prefer to not introduce it in core before the new filtering system exists.

**Pull request closed without merging**